### DISCUSSION TITLE
Preload all data to support firestore offline

### DISCUSSION THREAD

**Discussion by meliodev** [invertase/react-native-firebase#5194] [Q&A]
Hello guys,
I am testing my app on offline mode. And, sometimes not all data is cached when device turns offline. I know that It's normal.. As only data, which has been previously fetched from the server, is available when app turns offline. I am using a model where I fetch data on componentDidMount. For example, I fetch the list of Tasks only if user navigates to the screen listing the tasks.
Now I am thinking if preloading all data from firestore on app start up would be a good practice.  Or maybe somebody can propose a better solution.

**Comment by mikehardy** [comment]
Firestore actually just added the ability to do something similar - though I do not think it fits your exact usage pattern as it is targeted at data that *all* users would need at startup - "Bundles" https://firebase.google.com/docs/firestore/bundles

Which is just to say, the strategy certainly seems sound, and for certain use cases Firestore even has a new specific API for it (though we do not support it yet in react-native-firebase...)
